pkg/controllers/leaderworkerset_controller.go

```diff
@@ -122,27 +123,53 @@ func (r *LeaderWorkerSetReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return ctrl.Result{}, nil
 }
 
-func (r *LeaderWorkerSetReconciler) createHeadlessServiceIfNotExists(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet) error {
+func (r *LeaderWorkerSetReconciler) createMultipleHeadlessServices(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet) error {
+	if lws.Spec.NetworkConfig.SubdomainPolicy == leaderworkerset.SubdomainShared {
```

Comment:
Maybe you should check the nil case also. On newly created replica sets you can assume it’s defaulted. But you may need to support LWS that are coming from a case where this is not yet defaulted. In that case I think you’d want to fall back to the original case?

Reply:
Makes sense to me, I'll add the nil check. That does make me question if defaulting on the lws webhook is needed since the nil check does the same thing however.